pkgs/mkCatppuccinDerivation/package.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchCatppuccinPort,
+}:
+
+args:
+
+let
+  knownArguments = [
+    "fetchSubmodules"
+    "meta"
+    "portHash"
+    "portRev"
+  ];
+in
+
+stdenvNoCC.mkDerivation (
+  finalAttrs:
+
+  let
+    args' = if lib.isFunction args then args finalAttrs else args;
+  in
+
+  {
+    version = args'.version or builtins.substring 0 7 args'.portRev;
+
+    src = fetchCatppuccinPort {
+      port = finalAttrs.pname;
+      rev = args'.portRev;
+      hash = args'.portHash;
+      fetchSubmodules = args'.fetchSubmodules or false;
+    };
+
+    installPhase = ''
+      runHook preInstall
+
+      if [ -z "''${dontCatppuccinInstall:-}" ]; then
+        cp -r $src $out
+      fi
+
+      runHook postInstall
+    '';
+
+    meta = {
```

Comment:
might be worth adding a default `meta.maintainers` to prevent log spam for users that have `nixpkgs.config.showDerivationWarnings = [ "maintainerless" ];`